packages/autocomplete-core/src/getDefaultProps.ts

```diff
@@ -65,12 +65,10 @@ export function getDefaultProps<TItem>(
         .then((sources) =>
           sources.map((source) => ({
             ...source,
-            onSelect: (params) => {
+            onSelect(params) {
               source.onSelect(params);
               plugins.forEach((plugin) => {
-                if (plugin.onSelect) {
-                  plugin.onSelect(params);
-                }
+                plugin.subscribed?.onSelect?.(params);
```

Comment:
I don't understand why this is in an object with "subscribed" as name. What's the reason it's not directly on plugin?

Reply:
The plugin API must be identical to the Autocomplete API (minus some options like `openOnFocus`). The `onSelect` method is local to the sources, it's not on the Autocomplete instance.

What `plugin.subscribed.onSelect` does is that it's called whenever the `onSelect` method *from other sources* is called. Whereas [pseudocode] `plugin.getSources().onSelect` is called when this source's `onSelect` method is called.

I'm open to suggestions for the API, but we shouldn't allow to put top-level options on plugins that are not on the Autocomplete instance, otherwise we'll diverge.

Reply:
I see now, makes perfect sense :)